be/src/column/datum_convert.cpp

```diff
@@ -77,7 +82,7 @@ Status datum_from_string(TypeInfo* type_info, Datum* dst, const std::string& str
         }
         // If type is OLAP_FIELD_TYPE_CHAR, strip its tailing '\0'
         if (type == OLAP_FIELD_TYPE_CHAR) {
-            slice.size = strnlen(slice.data, slice.size);
+            slice.size = remove_trailing_spaces(slice.data, slice.size);
```

Comment:
If here we shouldn't call `strnlen`, I think we need to change many places used `strnlen` in storage.

Reply:
I use a different way of modification, treating CHAR as VARCHAR in Range to SeekTuple can avoid this problem